swabbie-core/src/main/kotlin/com/netflix/spinnaker/swabbie/discovery/DiscoveryActivated.kt

```diff
@@ -4,15 +4,25 @@ import com.netflix.appinfo.InstanceInfo
 import com.netflix.spinnaker.kork.eureka.RemoteStatusChangedEvent
 import org.slf4j.LoggerFactory
 import org.springframework.context.ApplicationListener
+import java.util.concurrent.atomic.AtomicBoolean
 
 /**
  * A component that starts doing something when the instance is up in discovery
  * and stops doing that thing when it goes down.
  */
-interface DiscoveryActivated : ApplicationListener<RemoteStatusChangedEvent> {
+open class DiscoveryActivated : ApplicationListener<RemoteStatusChangedEvent> {
+  private val up = AtomicBoolean()
+  open fun onDiscoveryUpCallback(event: RemoteStatusChangedEvent) {
```

Comment:
I think these should remain callbacks with no body, and the up/down logic should go in the `onApplicationEvent` function. That way, you can override these to provide specific logic without messing up the up/down indication or having to call `super`.

Reply:
Yes, the up/down can go in the onApplication body but I prefer these as overridable functions rather than properties though. The end result would be the same but reads better as functions.

Reply:
grr can't keep that without changing a bunch other stuff. Will keep them as getters